test/parallel_api/ranges/std_ranges_test.h

```diff
@@ -511,13 +530,13 @@ using  usm_span = usm_subrange_impl<T, std::span<T>>;
 
 #endif // TEST_DPCPP_BACKEND_PRESENT
 
-template<int call_id = 0, typename T = int, TestDataMode mode = data_in>
+template<int call_id = 0, typename T = int, TestDataMode mode = data_in, typename PolicySelector = AllPolicies>
 struct test_range_algo
 {
-    const int max_n = 10;
+    const int max_n = small_sz;
     void test_view(auto view, auto algo, auto& checker, auto... args)
     {
-        test<T, host_subrange<T>, mode>{max_n}(host_policies(), algo, checker, view, std::identity{}, args...);
+        test<T, host_subrange<T>, mode>{max_n}(PolicySelector{}, algo, checker, view, std::identity{}, args...);
```

Comment:
This approach requires to have all implementations of `PolicySelector` types but inside `test` function you still using only it's type check. I think this mean that better to pass `PolicySelector` type not through the instance but through additional template param.

This approach gives us ability to rewrite
```C++
struct AllPolicies{};
struct DeviceAndParPolicies{};
struct DevicePolicy{};
```

as
```C++
struct AllPolicies;
struct DeviceAndParPolicies;
struct DevicePolicy;
```